templates/template.go

```diff
@@ -0,0 +1,47 @@
+package templates
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+func replace(buf, newBytes []byte, offset, placeholderLength uint64) []byte {
+	output := append(buf[:offset], newBytes...)
+	return append(output, buf[(offset+placeholderLength):]...)
+}
+
+func inject(original []byte, offsets []uint64, values []interface{}) (result []byte, err error) {
+	result = original
+	if len(offsets) != len(values) {
+		err = fmt.Errorf("length of offsets %v does not match length of replacement values %v", len(offsets), len(values))
+		return
+	}
+
+	for i, value := range values {
+		decodedLength := 0
+
+		if valueAsUint, ok := value.(uint64); ok {
+			buffer := make([]byte, 8)
```

Comment:
largest # of bytes a uint64 could be is 8, so `8` is appropriate here, yeah?

Reply:
no, it can actually be larger. The encoding is different the "normal" one.

Reply:
What is its max size?

Reply:
`binary.MaxVarintLen64`